native-assets/fractional-NFT/project/contracts/f-NFT-contract/tests/functions/deposit.rs

```diff
@@ -0,0 +1,111 @@
+use crate::utils::{
+    interface::{deposit, total_assets, total_supply},
+    setup::{defaults, deploy, get_wallet_balance, setup_nft},
+};
+
+mod success {
+
+    use super::*;
+
+    #[tokio::test]
+    async fn creates_vault() {
+        let (_deployer, admin, f_nft_id, nft_id) = deploy().await;
+        let (nft_1, nft_2) = setup_nft(&admin.wallet, &admin.nft, nft_id).await;
+        let (vault_sub_id, vault_admin, share_asset1, _share_asset2, share_supply) =
+            defaults(&admin.wallet, nft_1, nft_2, f_nft_id);
+
+        assert_eq!(get_wallet_balance(&admin.wallet, &share_asset1).await, 0);
```

Comment:
I'd personally set up the tests so that the post condition variable can be asserted by using `pre-condition-var-name + 1` but this is fine